tests/capture/transforms/test_capture_dynamic_decompositions.py

```diff
@@ -0,0 +1,580 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit tests for the ``DynamicDecomposeInterpreter`` class."""
+# pylint:disable=protected-access,unused-argument, wrong-import-position, no-value-for-parameter, too-few-public-methods
+import pytest
+
+import pennylane as qml
+
+jax = pytest.importorskip("jax")
+
+from pennylane.capture.primitives import cond_prim, for_loop_prim, qnode_prim, while_loop_prim
+from pennylane.operation import Operation
+from pennylane.transforms.decompose import DynamicDecomposeInterpreter
+
+pytestmark = [pytest.mark.jax, pytest.mark.usefixtures("enable_disable_plxpr")]
+
+
+class SimpleCustomOp(Operation):
+    """Simple custom operation that contains a single gate in its decomposition"""
+
+    num_wires = 1
+    num_params = 0
+
+    def _init__(self, phi, wires, id=None):
+        super().__init__(phi, wires=wires, id=id)
+
+    @staticmethod
+    def _compute_plxpr_decomposition(wires):
+        qml.Hadamard(wires=wires)
```

Comment:
Shouldn't this `_compute_plxpr_decompositions` also accept `phi` as an argument?